app/models/oauth_client.rb

```diff
@@ -0,0 +1,25 @@
+class OauthClient < ApplicationRecord
+  validates :client_id, presence: true, uniqueness: true
+  validates :client_name, presence: true
+  validates :redirect_uri, presence: true
+
+  def valid_redirect_uri?(uri)
+    redirect_uri == uri
+  end
+
+  def create_authorization_code!(redirect_uri)
+    code = SecureRandom.hex(32)
+
+    Redis.current.setex(
+      "oauth_code:#{code}",
+      600,
+      {
+        client_id: client_id,
```

Comment:
Rails now allows you to implicitly pass variables into hashes & keyword arguments if they have the same name. So this can be:
```ruby
    Redis.current.setex(
      "oauth_code:#{code}",
      600,
      {
        client_id:,
        redirect_uri:,
        created_at: Time.current.iso8601
      }.to_json
    )
```